src/hotspot/cpu/aarch64/c1_LIRGenerator_aarch64.cpp

```diff
@@ -423,11 +423,11 @@ void LIRGenerator::do_ArithmeticOp_FPU(ArithmeticOp* x) {
 
   LIR_Opr reg = rlock(x);
   LIR_Opr tmp = LIR_OprFact::illegalOpr;
-  if (x->is_strictfp() && (x->op() == Bytecodes::_dmul || x->op() == Bytecodes::_ddiv)) {
+  if (x->op() == Bytecodes::_dmul || x->op() == Bytecodes::_ddiv) {
     tmp = new_register(T_DOUBLE);
```

Comment:
This variable `tmp` doesn't seem to be used. I guess this code originally came from x86 where it's passed through to `arithmetic_op_fpu()`.